packages/api/app/index.js

```diff
@@ -58,14 +61,36 @@ app.use((err, _req, res, _next) => {
 
 const server = awsServerlessExpress.createServer(app);
 
-const handler = (event, context) => {
+const handler = async (event, context) => {
+  const { dynamoTableNamesParameterName } = process.env;
+  if (!dynamoTableNamesParameterName) {
+    throw new MissingRequiredEnvVar('dynamoTableNamesParameterName environment variable is required for API Lambda');
+  }
+
+  const ssmClient = context.ssmClient || services.systemsManager();
+  const dynamoTableNamesParameter = await ssmClient.getParameter({
+    Name: dynamoTableNamesParameterName,
+  }).promise();
+  const dynamoTableNames = JSON.parse(dynamoTableNamesParameter.Parameter.Value);
+  // Set Dynamo table names as environment variables for Lambda
+  Object.keys(dynamoTableNames).forEach((tableEnvVarName) => {
+    process.env[tableEnvVarName] = dynamoTableNames[tableEnvVarName];
+  });
+
   // workaround to support multiValueQueryStringParameters
   // untill this is fixed: https://github.com/awslabs/aws-serverless-express/issues/214
   const modifiedEvent = {
     ...event,
     queryStringParameters: event.multiValueQueryStringParameters || event.queryStringParameters,
   };
-  awsServerlessExpress.proxy(server, modifiedEvent, context);
+  // see https://github.com/vendia/serverless-express/issues/297
+  return new Promise((resolve, reject) => {
```

Comment:
Version 4.0.0+ of the serverless-express package would allow us to use an async handler for this Lambda without a workaround, but would require other updates to our Lambda code:

https://github.com/vendia/serverless-express/blob/mainline/CHANGELOG.md#400-2021-02-01
https://github.com/vendia/serverless-express/blob/mainline/UPGRADE.md

And since this is a fix directed at the environment variable problem, I didn't want to blow up the scope here to do a bunch of API endpoint refactors